File: src/State.cpp
# include "State.hpp"

State::State(int i, std::string c, bool final_state) {
    id = i;
    chord = c;
    is_final_state = final_state;
}

int State::get_id() {
    return this->id;
}

std::string State::get_chord() {
    return this->chord;
}

bool State::get_is_final_state() {
    return this->is_final_state;
}

std::vector<Transition> State::get_transitions() {
    return this->transitions;
}

void State::set_id(int id) {
    this->id = id;
}

void State::set_chord(std::string chord) {
    this->chord = chord;
}

void State::set_is_final_state(bool is_final_state) {
    this->is_final_state = is_final_state;
}

void State::add_transition(Transition transition) {
    this->transitions.push_back(transition);
}

File: src/State.py
from Transition import Transition


class State:
	def __init__(self, state_id, chord, final_state):
		self.id = state_id
		self.transitions = []
		self.chord = chord
		self.is_final_state = final_state

    # para adicionar uma transição a lista de transições do estado
	def add_transition(self, transition: Transition):
		self.transitions.append(transition)
